src/AppContext.jsx

```diff
@@ -0,0 +1,295 @@
+import React, { useState, useEffect, createContext } from 'react';
+import Stopwatch from './components/timers/Stopwatch';
+import Countdown from './components/timers/Countdown';
+import XY from './components/timers/XY';
+import Tabata from './components/timers/Tabata';
+
+export const TimerContext = createContext({});
+
+export const AppContext = ({ children }) => {
+  const [timers, setTimers] = useState([]);
+  const [timersChanged, setTimersChanged] = useState(0);
+  const [appTimerAction, setAppTimerAction] = useState('');
+  const [appTimerIndex, setAppTimerIndex] = useState(-1);
+
+  // Stopwatch states
+  const [stopwatchTime, setStopwatchTime] = useState(0);
```

Comment:
If the configured timers list has two stopwatch timers in it this doesn't work, because they both use the same shared state variables. 

You can solve this by tracking a new state variable called `timerConfigurations`, which would be an array of timer configurations, and you can then make that available to your individual timer components to handle counting down the time.